lib/services/api/metric_rollups_service.rb

```diff
@@ -0,0 +1,28 @@
+module Api
+  class MetricRollupsService
+    REQUIRED_PARAMS = %w(resource_type capture_interval start_date).freeze
+
+    def self.query_metric_rollups(params)
+      validate_params(params)
+
+      params[:offset] ||= 0
```

Comment:
Ah! That's quite surprising to me - that this service would mutate some input params for the caller - perhaps this could be handled closer to where paging happens, or at least outside of this service?